workspaces/ballerina/inline-data-mapper/src/components/Diagram/Label/ArrayMappingOptionsWidget.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// tslint:disable: jsx-no-multiline-js
+import React from 'react';
+
+import { TypeKind } from '@wso2/ballerina-core';
+import { Codicon, Item, Menu, MenuItem } from '@wso2/ui-toolkit';
+import { css } from '@emotion/css';
+
+import { InputOutputPortModel, ValueType } from '../Port';
+import { genArrayElementAccessSuffix, getValueType } from '../utils/common-utils';
+import { MappingType } from '../Link';
+import { ExpressionLabelModel } from './ExpressionLabelModel';
+
+export const useStyles = () => ({
+    arrayMappingMenu: css({
+        pointerEvents: 'auto'
+    }),
+    itemContainer: css({
+        display: 'flex',
+        width: '100%',
+        alignItems: 'center'
+    }),
+});
+
+const a2aMenuStyles = {
+    backgroundColor: "var(--vscode-quickInput-background)",
+    boxShadow: "none",
+    padding: "0px",
+    border: "1px solid var(--vscode-debugIcon-breakpointDisabledForeground)"
+};
+
+const codiconStyles = {
+    color: 'var(--vscode-editorLightBulb-foreground)',
+    marginRight: '10px'
+}
+
+export interface ArrayMappingOptionsWidgetProps {
+    model: ExpressionLabelModel;
+}
+
+export function ArrayMappingOptionsWidget(props: ArrayMappingOptionsWidgetProps) {
+    const classes = useStyles();
+    const { link  } = props.model;
+    const pendingMappingType = link.pendingMappingType;
+
+    const sourcePort = link.getSourcePort() as InputOutputPortModel;
+    const targetPort = link?.getTargetPort() as InputOutputPortModel;
+    const valueType = getValueType(link);
+
+    const isValueModifiable = valueType === ValueType.Default
+        || valueType === ValueType.NonEmpty;
+    
+    const onClickMapArrays = async () => {
+        if (isValueModifiable) {
+        } else {
+        }
+    }
+
+    const onClickMapIndividualElements = async () => {
+        
+    };
+
+    const onClickMapArraysAccessSingleton = async () => {
+       
+    };
+
+    const onClickAggregateArray = async () => {
+
+    };
+
+    const onClickMapUsingCustomFunction = async () => {
+    };
+
+    const onClickMapUsingCollectClause = async (collectFunction: string) => {
+
+    }
```

Comment:
Let's remove the unused functions and unsupported mapping options and add them once the feature is ready.